### PR TITLE
Check authoritative nameservers for a domain

### PR DESCRIPTION
**Pull request by parkr**
DNS caching causes all sorts of issues, because checking the domain once caches the record
on many layers between the querying host and the authoritative nameservers.

To get around caching, simply ask the authoritative nameservers for the domain, bypassing
the DNS in the local network and all networks in between.

### CODE REVIEW

File: lib/github-pages-health-check.rb [hunk 1]
```diff
@@ -48,6 +48,31 @@ module HealthCheck
       }
     }.freeze
 
+    class << self
+      def default_resolver
```
Comment (benbalter) [inline lib/github-pages-health-check.rb]: Is there value in memoizing this?

File: lib/github-pages-health-check.rb [hunk 2]
```diff
@@ -48,6 +48,31 @@ module HealthCheck
...
+          next answer.nsdname if answer.type == Dnsruby::Types::NS
+        end.compact
+      end
+
+      def build_resolver(domain)
+        Dnsruby::Resolver.new(
+          nameservers: nameservers_for_domain(domain),
+          retry_times: 2,
+          query_timeout: 2,
+          dnssec: true,
```
Comment (benbalter) [inline lib/github-pages-health-check.rb]: To DRY things up, perhaps move the default options to a class-level constant/hash and just merge the nameservers in here?

**Comment by nickcannariato** [comment]
cc/ @github/support-pages-ombuds @shawnajean for visibility.

**Comment by parkr** [comment]
@benbalter Completely rewrote this. Essentially it only queries the authoritative DNS for `A`, `CAA`, and `MX` records. All other types of records will hit the default DNS resolver.

**Comment by benbalter** [comment]
Looks great. Can/should the new class have tests?

**Comment by parkr** [comment]
> Looks great. Can/should the new class have tests?

LOL, good poont. Added!

**Comment by parkr** [comment]
> So nice I approved it twice.

Thank you!! ❤️

**Pull request merged**